src/main/java/mate/academy/carsharing/notification/NotificationTelegramBot.java

```diff
@@ -0,0 +1,107 @@
+    @Value("${bot.name}")
+    private String botName;
+
+    @Value("${bot.key}")
+    private String botToken;
+
+    private final TelegramUserService telegramUserService;
+    private final TelegramUserRepository telegramUserRepository;
+
+    public NotificationTelegramBot(@Value("${bot.key}") String botToken,
+                                   TelegramUserService telegramUserService,
+                                   TelegramUserRepository telegramUserRepository) {
+        this.botToken = botToken;
+        this.telegramUserService = telegramUserService;
+        this.telegramUserRepository = telegramUserRepository;
+    }
+
+    @Override
+    public String getBotUsername() {
+        return botName;
+    }
+
+    @Transactional
+    @Override
+    public void onUpdateReceived(Update update) {
+        String chatId = String.valueOf(update.getMessage().getChatId());
+        String messageText = update.getMessage().getText();
+
+        System.out.println("Received message from chatId - " + chatId + " is: " + "''"
+                + messageText + "''");
+
+        List<TelegramUserInfo> userInfoList = telegramUserRepository.findByChatIdWithRoles(chatId);
+
+        if (!userInfoList.isEmpty()) {
+            telegramUserRepository.deleteByChatId(chatId);
+        }
+
+        if (update.hasMessage() && update.getMessage().hasText()) {
+            String message = update.getMessage().getText();
+            if (message.equals("/start")) {
+                // Відправляємо стартове повідомлення і додаємо новий запис
+                startingMessage(chatId);
+            } else {
+                validateUser(message, chatId);
+            }
+        }
+    }
+
+    private void startingMessage(String chatId) {
+        String message = """
+                This bot will inform you about:
+                * Rentals creation
+                * Payments creation
+                * Successful payments
+                * Overdue rentals
+
+                Please, enter your e-mail to authorize""";
+        sendMessage(message, chatId);
+        TelegramUserInfo newUserInfo = new TelegramUserInfo();
+        newUserInfo.setChatId(chatId);
+        telegramUserRepository.save(newUserInfo);
+    }
+
+    private void validateUser(String email, String chatId) {
+        try {
+            telegramUserService.save(email, chatId);
+            sendMessage("Congratulations, the authorization was successful!", chatId);
+        } catch (Exception e) {
+            sendMessage("To start authorization, enter '/start'. ", chatId);
+        }
+    }
+
+    public void sendMessage(String value, String chatId) {
+        if (chatId == null) {
+            return;
+        }
+
+        SendMessage message = new SendMessage(chatId, value);
+        try {
+            execute(message);
+        } catch (TelegramApiException e) {
+            throw new TelegramBotMessageException("Couldn't send a message: " + value, e);
+        }
+    }
```

Comment:
You can create service and extrazt some logic to this service, can create some handlers for commands